index.js

```diff
@@ -21,15 +21,21 @@ const validationErrors = async (db, collectionName, { doc, err }) => {
 
 const explainSchemaErrors = (incomingDb, options = {}) => {
   const db = incomingDb;
-  const { onError } = options;
+  const { onError, includeValidationInError } = options;
   if (onError) {
     db.onValidationError = onError;
   }
-  const explainValidationError = async (...args) => {
+  const explainValidationErrorLogic = async (...args) => {
     const { valid, errors } = await validationErrors(...args);
-    if (!valid) {
+    if (!valid && db.onValidationError) {
       db.onValidationError(errors);
     }
+    return errors;
+  };
+  // If returning validation errors, wait for them
+  const explainValidationError = async (...args) => {
+    if (includeValidationInError) return explainValidationErrorLogic(...args);
+    else explainValidationErrorLogic(...args);
```

Comment:
If `includeValidationInError` is true it will lead to a slight additional delay for that particular request as it will now wait for the error to be returned. If `includeValidationInError` is false it will not wait (like it was originally).

Reply:
Yes, my bad here - explicit return sounds good